### DISCUSSION TITLE
Reactive REST Client: Possible regression in 3.5.2

### DISCUSSION THREAD

**Discussion by martin-kofoed-jyskebank-dk** [quarkusio/quarkus#37220] [Q&A] [answered]
Hi,

I'm a bit reluctant to file this as an issue, because it's related to a bug fix in 3.5.2. 

I have a classic filter `@Provider` that implements all three filter interfaces in the same class. After bumping from 3.5.1 to 3.5.2, the `ClientRequestFilter` is no longer invoked for reactive REST clients. Workaround is to explicitly register the filter class on the client with `@RegisterProvider`. Another workaround would probably be to implement the client filter in a separate class.

I believe it is because of this specific change for 3.5.2:

https://github.com/quarkusio/quarkus/pull/37006/files

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
cc @famod

**Reply by famod** [reply]
Good catch! The current impl doesn't expect such a "mixed" class.

I can have a look later this week.

**Reply by famod** [reply]
@martin-kofoed-jyskebank-dk can you create an issue for that please? Thanks!

**Reply by martin-kofoed-jyskebank-dk** [reply, marked as answer]
@famod I took the liberty to submit a PR instead. Feel free to discard it in favor of your own implementation. I can also do a formal issue if that's better? 

https://github.com/quarkusio/quarkus/pull/37268
